src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,414 @@
+        }
+
+        return client.newCall(POSTJson(queryAPI, data))
+            .asObservableSuccess()
+            .map { response ->
+                parseMangaList(response, "recentUpdate")
+            }
+    }
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        if (query.startsWith(PREFIX_ID_SEARCH)) {
+            val id = query.removePrefix(PREFIX_ID_SEARCH)
+
+            val data = buildJsonObject {
+                put("operationName", "comicById")
+                putJsonObject("variables") {
+                    put("comicId", id)
+                }
+                put("query", "query comicById(\$comicId: ID!) {\n  comicById(comicId: \$comicId) {\n    id\n    title\n    status\n    year\n    imageUrl\n    authors {\n      id\n      name\n      __typename\n    }\n    categories {\n      id\n      name\n      __typename\n    }\n    dateCreated\n    dateUpdated\n    views\n    favoriteCount\n    lastBookUpdate\n    lastChapterUpdate\n    __typename\n  }\n}")
+            }
+
+            return client.newCall(POSTJson(queryAPI, data))
+                .asObservableSuccess()
+                .map { response ->
+                    parseMangaIDList(response)
+                }
+        } else {
+            val data = buildJsonObject {
+                put("operationName", "searchComicAndAuthorQuery")
+                putJsonObject("variables") {
+                    put("keyword", query)
+                }
+                put("query", "query searchComicAndAuthorQuery(\$keyword: String!) {\n  searchComicsAndAuthors(keyword: \$keyword) {\n    comics {\n      id\n      title\n      status\n      year\n      imageUrl\n      authors {\n        id\n        name\n        __typename\n      }\n      categories {\n        id\n        name\n        __typename\n      }\n      dateUpdated\n      monthViews\n      views\n      favoriteCount\n      lastBookUpdate\n      lastChapterUpdate\n      __typename\n    }\n    authors {\n      id\n      name\n      chName\n      enName\n      wikiLink\n      comicCount\n      views\n      __typename\n    }\n    __typename\n  }\n}")
+            }
+
+            return client.newCall(POSTJson(queryAPI, data))
+                .asObservableSuccess()
+                .map { response ->
+                    parseMangaList(response, "searchComicsAndAuthors")
+                }
+        }
+    }
+
+    /**
+     * 解析漫畫詳情
+     */
+    private fun parseMangaDetails(response: Response): SManga {
+        val jsonData = response.body.string()
+        val jsonObj = JSONObject(jsonData).getJSONObject("data").getJSONObject("comicById")
+
+        return SManga.create().apply {
+            url = "https://komiic.com/comic/${jsonObj.getString("id")}"
```

Comment:
save only the id in db and construct the urls in code (`request` and `getMangaUrl`)